### PR TITLE
Move OtherName and SANS functions to s/s

### PR DESCRIPTION
**Pull request by haydentherapper**
These were used by both Cosign and Fulcio, and will now be used by Rekor, so they should be in the shared library.

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

### CODE REVIEW

File: pkg/cryptoutils/sans.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			// skip all GeneralName fields except OtherName
+			if v.Tag != 0 {
+				continue
+			}
+
+			var other OtherName
+			_, err := asn1.UnmarshalWithParams(v.FullBytes, &other, "tag:0")
+			if err != nil {
+				return "", fmt.Errorf("could not parse requested OtherName SAN: %w", err)
+			}
```
Comment (bobcallaway) [inline pkg/cryptoutils/sans.go:113]: nit (only fix if you're respinning this for some reason):
```suggestion
			if _, err := asn1.UnmarshalWithParams(v.FullBytes, &other, "tag:0"); err != nil {
				return "", fmt.Errorf("could not parse requested OtherName SAN: %w", err)
			}
```

File: test/cert_utils.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	certBytes, err := x509.CreateCertificate(rand.Reader, template, parent, pub, priv)
+	if err != nil {
+		return nil, err
+	}
+
+	cert, err := x509.ParseCertificate(certBytes)
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
```
Comment (bobcallaway) [inline test/cert_utils.go:63]:
```suggestion
	return x509.ParseCertificate(certBytes)
```

**Review by bobcallaway — COMMENTED** [review]
lgtm minus nits

**Comment by haydentherapper** [comment]
@bobcallaway ready for LGTM

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**